packages/core-app-api/src/apis/implementations/FetchApi/FetchMiddlewares.ts

```diff
@@ -56,14 +56,16 @@ export class FetchMiddlewares {
    *
    * The header injection only happens on allowlisted URLs. Per default, if the
    * `config` option is passed in, the `backend.baseUrl` is allowlisted, unless
-   * the `urlPrefixAllowlist` option is passed in, in which case it takes
-   * precedence. If you pass in neither config nor an allowlist, the middleware
-   * will have no effect.
+   * the `urlPrefixAllowlist` or `allowUrl` options are passed in, in which case
+   * they take precedence. If you pass in neither config nor an
+   * allowlist/callback, the middleware will have no effect since effectively no
+   * request will match the (nonexistent) rules.
    */
   static injectIdentityAuth(options: {
     identityApi: IdentityApi;
     config?: Config;
     urlPrefixAllowlist?: string[];
+    allowUrl?: (url: string) => boolean;
```

Comment:
@Rugvip ^ To allow for more flexibility, and to let the mock fetch api say `() => true` here. So I think one last thing to settle is the awkward interface here. I want to remove `urlPrefixAllowlist` because it's an awkward (and not in use in the code base yet) subset of what `allowUrl` can do.

Or alternatively, invent a new matcher interface that you pass in here, and then have like `IdentityAuthUrlMatcher.fromConfig()`/`.fromCallback`/`.fromAllowlist` and stuff here but it kinda kills the simplicity...

Reply:
yeah I think this + removing `urlPrefixAllowlist` is the way to go

Reply:
"This" being the simple callback in this PR? :)